mmdet3d/models/necks/dla_neck.py

```diff
@@ -0,0 +1,204 @@
+                padding=up_kernel_size // 2,
+                output_padding=0,
+                groups=out_channel,
+                bias=False)
+            setattr(self, 'proj_' + str(i), proj)
+            setattr(self, 'up_' + str(i), up)
+            setattr(self, 'node_' + str(i), node)
+
+    def forward(self, layers, start_level, end_level):
+        """Forward function.
+
+        Args:
+            layers (list[torch.Tensor]): Features from multiple layers.
+            start_level (int): Start layer for feature upsampling.
+            end_level (int): End layer for feature upsampling.
+        """
+        for i in range(start_level + 1, end_level):
+            upsample = getattr(self, 'up_' + str(i - start_level))
+            project = getattr(self, 'proj_' + str(i - start_level))
+            layers[i] = upsample(project(layers[i]))
+            node = getattr(self, 'node_' + str(i - start_level))
+            layers[i] = node(layers[i] + layers[i - 1])
+
+
+class DLAUp(nn.Module):
+    """DLAUp module for multiple layers feature extraction and fusion.
+
+    Args:
+        start_level (int): The start layer.
+        channels (List[int]): List of input channels of multi-scale
+            feature map.
+        scales(List[int]): List of scale of different layers' feature.
+        in_channels (NoneType): List of input channels of different scales.
+            Default: None.
+        norm_cfg (dict): Config dict for normalization layer. Default: None.
+    """
+
+    def __init__(self,
+                 start_level,
+                 channels,
+                 scales,
+                 in_channels=None,
+                 norm_cfg=None):
+        super(DLAUp, self).__init__()
+        self.start_level = start_level
+        if in_channels is None:
+            in_channels = channels
+        self.channels = channels
+        channels = list(channels)
+        scales = np.array(scales, dtype=int)
+        for i in range(len(channels) - 1):
+            j = -i - 2
+            setattr(
+                self, 'ida_{}'.format(i),
+                IDAUp(channels[j], in_channels[j:], scales[j:] // scales[j],
+                      norm_cfg))
+            scales[j + 1:] = scales[j]
+            in_channels[j + 1:] = [channels[j] for _ in channels[j + 1:]]
+
+    def forward(self, layers):
+        """Forward function.
+
+        Args:
+            layers (tuple[torch.Tensor]): Features from multi-scale layers.
+
+        Returns:
+            tuple[torch.Tensor]: Up-sampled features of different layers.
+        """
+        outs = [layers[-1]]
+        for i in range(len(layers) - self.start_level - 1):
+            ida = getattr(self, 'ida_{}'.format(i))
+            ida(layers, len(layers) - i - 2, len(layers))
+            outs.insert(0, layers[-1])
+        return outs
+
+
+@NECKS.register_module()
+class DLANeck(nn.Module):
+    """DLA Neck.
+
+    Args:
+        in_channels (list[int]): List of input channels of multi-scale
```

Comment:
`optional`

Reply:
If the default value if too long (e.g. a list of 6 number in this case), we can omit it